app/questionnaire/router.py

```diff
@@ -38,27 +44,29 @@ def get_next_location_url(self, location, routing_path):
         current_block_type = self._schema.get_block(location.block_id)['type']
         last_block_location = routing_path[-1]
         last_block_type = self._schema.get_block(last_block_location.block_id)['type']
-        section_id = self._schema.get_section_id_for_block_id(location.block_id)
+
+        section_key = (location.section_id, location.list_item_id)
 
         if (
             self._schema.is_hub_enabled()
             and location.block_id == last_block_location.block_id
-            and section_id in self._progress_store.completed_section_ids
+            and section_key in self._progress_store.completed_section_keys
```

Comment:
Rather than passing in a tuple here, should progress_store have an is_section_completed method which we can pass a section_id and list_item_id?

Reply:
Done.

Reply:
Yeah. To me `section_key` is an implementation detail of the store, we shouldn't be exposing it if possible.